Firestore/Example/Tests/Local/FSTPersistenceTestHelpers.mm

```diff
@@ -41,10 +41,8 @@
 @implementation FSTPersistenceTestHelpers
 
 + (FSTLocalSerializer *)localSerializer {
-  // This owns the DatabaseIds since we do not have FirestoreClient instance to own them.
-  static DatabaseId database_id{"p", "d"};
-
-  FSTSerializerBeta *remoteSerializer = [[FSTSerializerBeta alloc] initWithDatabaseID:&database_id];
+  DatabaseId database_id{"p", "d"};
+  FSTSerializerBeta *remoteSerializer = [[FSTSerializerBeta alloc] initWithDatabaseID:database_id];
```

Comment:
nit: Is there a reason the previous file (FSTLocalSerializerTests.mm) has:
```
  DatabaseId databaseId("p", "d");
  self.remoteSerializer = [[FSTSerializerBeta alloc] initWithDatabaseID:std::move(databaseId)];
```

and here we have:
```
  DatabaseId database_id{"p", "d"};
  FSTSerializerBeta *remoteSerializer = [[FSTSerializerBeta alloc] initWithDatabaseID:database_id];
```

It doesn't really matter, but should we try to be consistent?

FWIW- I think now that we don't have to get a pointer to the DatabaseId, the construction could just be inlined (drop the "database_id" intermediate variable) and that would probably be cleaner. But if you want to keep this PR smaller / mechanical, I'm fine with that too.

Reply:
I assume you're asking about the `DatabaseId` declaration more than the rest (`FSTLocalSerializerTests` retains the `remoteSerializer` for direct access where this doesn't need that).

I think the issue is that some time ago some porting PRs introduced C++ names into Objective-C sources with the idea that we'd eventually be renaming to that anyway. This has led to the mish-mash you're observing.

I've taken your suggestion and removed the locals. Here and throughout the PR. See 370ffade7488d801e52bf50247c1aae6c9fc1f2d.